### DISCUSSION TITLE
Some questions about the game semantics

### DISCUSSION THREAD

**Discussion by golddranks** [tonarino/bombercrab-player#4] [Q&A] [answered]
1) I see that there is an `Object::is_solid` method defined in `bomber_lib/src/world.rs`, which returns true for bombs and crates. I expect this to mean that similar to walls, you can't enter coordinates with bombs or crates. However, there is also `Action::DropBomb` with the docstring `Place a bomb at your current location.`. Would this mean that you can stand on a coordinate with a bomb, and walk away from that, but you can't enter such a coordinate?
2) There is the field `Option<Enemy>` in the surroundings data. Am I correct to assume that there can be only a single player per a coordinate, and you can't enter a coordinate with another player? What happens if two players try to enter the same coordinate on the same tick?
3) The method `max_count_per_player` seems to document the maximum powerups, but are there information about the default/starting powerups available?
4) How much is the initial value of `Bomb::fuse_remaining`? 
5) Is the size of the world documented? Are the borders of the world infinite, blocked by walls or perhaps wrap-around?
6) Does the scoreboard function by individual players? (I.e. team sum scores are not available?)

**Comment by PabloMansanet** [comment, marked as answer]
Hello @golddranks !

Our intention was to leave the answer to these things a mystery until the event, but your questions are great and it's pretty cool to see this level of analysis and interest, so let me give some vague hints without spoiling much :)

1. You're on the right track. In this way, we try to make the behaviour a lot like classic Bomberman games.
2. Players cannot occupy the same spot. For other behaviours such as crossing paths, you'll have to play and find out! As for both players attempting to enter the same tile, only one will make it there, though I wouldn't rely on either in particular.
3. No such information yet ;)
4. Gee, it would be terrible to find out the hard way, wouldn't it? ;)
5. There's no wraparound, but other than that the particulars of the map are up in the air. Or should I say map(s)...?
6. Teams are entirely cosmetic: they only affect the color of your character, and what shows up on the victory screen. Victory points are tallied individually, and it's up to the teams to choose whether to collaborate or hinder each other! As you have probably seen in the Enemy struct, you get a chance to query for other player's team allegiances, which you can use to your advantage if you choose to work together.

See you soon!

**Reply by golddranks** [reply]
Thank you! Yeah, I totally understand that you guys want to keep stuff a mystery until the event! Thanks for the hints anyway!
